pkg/tests/testdata/resource-ordering/Pulumi.yaml

```diff
@@ -0,0 +1,17 @@
+name: aws-yaml
+runtime: yaml
+resources:
+  alb:
```

Comment:
I'd like to see this use two default providers, from two different packages _and_ show that we can have one default provider depend on the output of a resource from the other package. Something like the following:
resA: type: pkg1:idx:typ
resB: type: pkg2:idx:typ
provA: type: pulumi:providers:pkg1
provB: type: pulumi:providers:pkg2
   properties: input: resA.output
   
Should sort as [provA, resA, provB, resB]

Reply:
WOO did it, here is the output, ill try to add a test for topological sort as well:

  Type                              Name                Plan       Info
 +   pulumi:pulumi:Stack               aws-yaml-teststack  create     2 warnings; 1 message
 +   ├─ pulumi:providers:aws           provider            create
 +   ├─ aws:alb:LoadBalancer           alb                 create
 +   ├─ pulumi:providers:testprovider  testProvider        create
 +   └─ testprovider:index:Echo        echo                create

Reply:
I made it a child PR just in case we wanna roll back one at a time and not squash them, lmk if you prefer to just combine them